### PR TITLE
Feature/sprockets 23

### LINKED ISSUE #203: undefined method `each_entry' for #<Sprockets::Environment:0x007fb129769378>
```
konacha (3.3.0) lib/konacha.rb:40:in `block in spec_paths'
Rails.application.assets.each_entry(root).find_all { |pathname|
```

It seems like each_entry was removed from sprockets a while ago:

https://github.com/rails/sprockets/commit/6c7b2dc045fbf7b77fe92246f47ae3eaf3c9c475

### PR DESCRIPTION
**Pull request by jonhyman**
Fixes #203

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Awesome!

The build errors on Travis are all `LoadError` on tilt, stemming from the fact that Sprockets 2 had a dependency on tilt that konacha was implicitly depending on and Sprockets 3 no longer has that dependency. Perhaps we can remove the ` || Tilt[pathname]` condition in the second case. I don't recall exactly what that case is for.

File: konacha.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ the asset pipeline and engines.}
 
   gem.add_dependency "railties", ">= 3.1", "< 5"
   gem.add_dependency "actionpack", ">= 3.1", "< 5"
-  gem.add_dependency "sprockets", "~> 2"
+  gem.add_dependency "sprockets"
```
Comment (gabebw) [inline konacha.gemspec]: "> 2", "< 4"?

Comment (jonhyman) [inline konacha.gemspec]: Will do `>= 2, <4`

**Comment by jonhyman** [comment]
So I tried removing Tilt for Sprockets 3 but lots of tests then fail. I also did not run the tests after upgrading to sprockets 3 and this spec fails. Let me work on this a bit more later today, I have to run to a few meetings.

> 1) Konacha.spec_paths with additional spec directories has specs from app/sections
     Failure/Error: subject.should include("my_section/my_section_spec.js.coffee")
       expected

**Comment by jonhyman** [comment]
@jfirebaugh I added a dependency for tilt and then fixed the spec for Sprockets 3.

File: spec/konacha_spec.rb [hunk 2]
```diff
@@ -104,8 +104,12 @@ def self.===(path)
         begin
           spec_dir = Konacha.config.spec_dir
           Konacha.configure {|c| c.spec_dir = ["spec/javascripts", "app/sections"]}
-          example.run
           Rails.application.config.assets.paths << Rails.root.join("app/sections").to_s
+          # For Sprockets 3, adding to config.assets.paths does not work
+          if Rails.application.assets.respond_to?(:append_path)
+            Rails.application.assets.append_path(Rails.root.join("app/sections").to_s)
+          end
```
Comment (jfirebaugh) [inline spec/konacha_spec.rb]: Should the existing `assets.paths <<` case be in an `else` clause here?

Comment (jonhyman) [inline spec/konacha_spec.rb]: I updated it to be in one.

**Comment by jfirebaugh** [comment]
Thanks, nice work! Committed as 82928daf8a2a0917c74eca2f475ba476e5cc03d9.

**Comment by jonhyman** [comment]
Awesome, thanks!

**Comment by gabebw** [comment]
Oh man, thank you! This is great news!

**Pull request closed without merging**